yarn-project/txe/src/oracle/txe_oracle.ts

```diff
@@ -219,11 +220,36 @@ export class TXE implements TypedOracle {
     return this.txeDatabase.addAuthWitness(authWitness.requestHash, authWitness.witness);
   }
 
-  async addNullifiers(contractAddress: AztecAddress, nullifiers: Fr[]) {
+  async addPublicDataWrites(writes: PublicDataUpdateRequest[]) {
```

Comment:
public executor no longer commits to worldStateDB, so we need to manually perform tree insertions for public data writes here

Reply:
TY David! LGTM, provided the current TXE test suite passes (which it does!)

Seems like we don't have to build the `CombinedConstantData` in its entirety now, is this going to stay that way? If it is, I'll leave just the `GlobalVariables` struct in  `executePublicFunction` in an upcoming refactor PR, should clean it up quite nicely.

Reply:
Hmm I think it will stay that way, at least for the TXE's interface to the executor